math/mathmore/src/complex_quartic.h

```diff
@@ -222,7 +215,7 @@ gsl_poly_complex_solve_quartic (double a, double b, double c, double d,
        * mt=3 : 2 real roots
        */
       // when disc == 0  2 roots are identicals
-      if (0 >= disc)
+      if (1e-16 >= disc)
```

Comment:
Please avoid magic numbers, especially if you use the same magic number twice in a file. I should then be defined as a `constexpr` in the file. Also, consider using [numeric_limits::epsilon](https://en.cppreference.com/w/cpp/types/numeric_limits/epsilon} for this, or a factor thereof.

Reply:
`epsilon` is very likely the wrong choice because `disc` is the result of multiple operations, so the accumulated error can easily be larger than that. Regarding the use of magic numbers, I would argue that the entire file is already full of them...

Reply:
Fair enough, then let's keep this fix as it is! At least I tried to encourage some general code improvements, but I agree that this is not necessarily the place for it :slightly_smiling_face:

Reply:
> `epsilon` is very likely the wrong choice because `disc` is the result of multiple operations, so the accumulated error can easily be larger than that. Regarding the use of magic numbers, I would argue that the entire file is already full of them...

That's why I suggested a factor of `epsilon`

Reply:
I added a constant `ZeroTolerance` and also increased it to `1e-15` because I noticed that `epsilon` is around `2e-16`. The reason it still works is because that epsilon is around the value 1, but it also shifts with different values. Maybe we should consider even a higher tolerance...